src/lib/connectors/magicConnectConnector.ts

```diff
@@ -21,10 +22,14 @@ import { getAddress } from 'ethers/lib/utils.js'
 export interface MagicConnectorOptions {
   apiKey: string
   magicSdkConfiguration?: MagicSDKAdditionalConfiguration
+  networks?: EthNetworkConfiguration[]
```

Comment:
Cool looks good now. also, instead of taking in networks separately, can't we directly pass it to magicSdkConfiguration.network ?

Reply:
Unfortunately, no, this is the issue with magic that does not accept a list of networks, and `MagicSDKAdditionalConfiguration` is a type from Magic.
We could have some kind of hack like `MagicSDKAdditionalConfiguration & { networks?: EthNetworkConfiguration[] }` but that might complicate the rest of the codebase.
Something that could be done tho is to "remove" the `network` from the `magicSdkConfiguration` with something like `Omit<MagicSDKAdditionalConfiguration, 'network'>` and only keep the `networks` but that would create a breaking change.

Reply:
I see, I think we can do a a type extension and allow users to pass either the object directly as original magic api

```ts
magicSdkConfiguration: {
  network: {
    rpcUrl: "https://your-ethereum-rpc-url",
    chainId: 1,
  },
}
```
or with an array 
```ts
magicSdkConfiguration: {
  network: [
  { chainId: 1, rpcUrl: 'https://your-ethereum-rpc-url' }
  { chainId: 137, rpcUrl: 'https://your-polygon-rpc-url' }
  ],
}
``` 

Tho this is just options for our connector, we can just pass correct chain item to magic constructor on switch chain so it still only gets single item. 

This way we can just pass `magicSdkConfiguration.network[0]` if its an array or `magicSdkConfiguration.network` as a whole, which should be compatible with MagicAuthConnector as well. that way users have only one place to add their networks.  

And we can throw error on switchChain method incase the `magicSdkConfiguration.network` is not an array. 

But i believe the current implementation is good too, since the network switch is something we are implementing, makes sense to extract it apart from `magicSdkConfiguration`.

Reply:
Lets go this 😄 its good for merge i think

Reply:
Your solution could definitely work, just the naming `magicSdkConfiguration` would be weird as the official documentation https://magic.link/docs/connect/wallet-api-reference/javascript-client-sdk has no mention of that, so in that case, this should not be named `magicSdkConfiguration` in my opinion.

Reply:
Yes thats true. my thought as well !!